crates/holochain/src/sweettest/sweet_consistency.rs

```diff
@@ -32,60 +32,174 @@ impl DurationOrSeconds {
 pub async fn await_consistency<'a, I: IntoIterator<Item = &'a SweetCell>>(
     timeout: impl Into<DurationOrSeconds>,
     all_cells: I,
-) -> ConsistencyResult {
-    await_consistency_advanced(timeout, (), all_cells.into_iter().map(|c| (c, true))).await
+) -> Result<(), String> {
+    #[allow(clippy::type_complexity)]
+    let all_cell_dbs: Vec<(AgentPubKey, DbRead<DbKindDht>)> = all_cells
+        .into_iter()
+        .map(|c| (c.agent_pubkey().clone(), c.dht_db().clone().into()))
+        .collect();
+    let all_cell_dbs: Vec<_> = all_cell_dbs.iter().map(|c| (&c.0, &c.1)).collect();
+    await_op_integration(&all_cell_dbs[..], timeout.into().into_duration()).await
 }
 
-/// Wait for all cells to reach consistency
-#[cfg_attr(feature = "instrument", tracing::instrument(skip_all))]
-pub async fn await_consistency_conditional<'a, I: IntoIterator<Item = &'a SweetCell>>(
-    timeout: impl Into<DurationOrSeconds>,
-    conditions: impl Into<ConsistencyConditions>,
-    all_cells: I,
-) -> ConsistencyResult {
-    await_consistency_advanced(
-        timeout,
-        conditions,
-        all_cells.into_iter().map(|c| (c, true)),
-    )
-    .await
+#[derive(Clone, Debug)]
+struct DhtOpRow {
+    hash: DhtOpHash,
+    op_type: DhtOpType,
+    action_seq: u32,
+    author: AgentPubKey,
+    when_integrated: Option<Timestamp>,
 }
 
-/// Wait for all cells to reach consistency,
-/// with the option to specify that some cells are offline.
-///
-/// Cells paired with a `false` value will have their authored ops counted towards the total,
-/// but not their integrated ops (since they are not online to integrate things).
-/// This is useful for tests where nodes go offline.
-#[cfg_attr(feature = "instrument", tracing::instrument(skip_all))]
-pub async fn await_consistency_advanced<'a, I: IntoIterator<Item = (&'a SweetCell, bool)>>(
-    timeout: impl Into<DurationOrSeconds>,
-    conditions: impl Into<ConsistencyConditions>,
-    all_cells: I,
-) -> ConsistencyResult {
-    #[allow(clippy::type_complexity)]
-    let all_cell_dbs: Vec<(
-        AgentPubKey,
-        DbRead<DbKindAuthored>,
-        Option<DbRead<DbKindDht>>,
-    )> = all_cells
-        .into_iter()
-        .map(|(c, online)| {
-            (
-                c.agent_pubkey().clone(),
-                c.authored_db().clone().into(),
-                online.then(|| c.dht_db().clone().into()),
-            )
-        })
-        .collect();
-    let all_cell_dbs: Vec<_> = all_cell_dbs
-        .iter()
-        .map(|c| (&c.0, &c.1, c.2.as_ref()))
-        .collect();
-    wait_for_integration_diff(
-        &all_cell_dbs[..],
-        timeout.into().into_duration(),
-        conditions.into(),
-    )
-    .await
+/// Wait for all cell envs to reach consistency, meaning that every op
+/// published by every cell has been integrated by every node.
+async fn await_op_integration(
```

Comment:
This used to be `wait_for_integration_diff`. Moved here and renamed and rewritten.